cnv_common_tasks.wdl

```diff
@@ -0,0 +1,598 @@
+task PreprocessIntervals {
+    File? intervals
+    File? blacklist_intervals
+    File ref_fasta
+    File ref_fasta_fai
+    File ref_fasta_dict
+    Int? padding
+    Int? bin_length
+    File? gatk4_jar_override
+
+    # Runtime parameters
+    String gatk_docker
+    Int? mem_gb
+    Int? disk_space_gb
+    Boolean use_ssd = false
+    Int? cpu
+    Int? preemptible_attempts
+
+    Int machine_mem_mb = select_first([mem_gb, 2]) * 1000
+    Int command_mem_mb = machine_mem_mb - 500
+
+    # Determine output filename
+    String filename = select_first([intervals, "wgs"])
+    String base_filename = basename(filename, ".interval_list")
+
+    command <<<
+        set -e
+        export GATK_LOCAL_JAR=${default="/root/gatk.jar" gatk4_jar_override}
+
+        gatk --java-options "-Xmx${command_mem_mb}m" PreprocessIntervals \
+            ${"-L " + intervals} \
+            ${"-XL " + blacklist_intervals} \
+            --sequence-dictionary ${ref_fasta_dict} \
```

Comment:
@samuelklee do we need a dictionary if we have the reference?

Reply:
Hmm, I guess we don't---good catch.  My guess is that a previous version of the tool did not have the N-masking functionality (for which a reference is required) and thus only required the sequence dictionary for resolving input intervals that lacked one.

Let's address any issues with the original WDLs at the same time ashttps://github.com/broadinstitute/gatk/issues/6502.  We should make sure that the somatic and germline WDLs are both changed.